server/files/api_views.py

```diff
@@ -21,14 +22,24 @@ def perform_destroy(self, instance):
 
     def create(self, request):
         (metadata, file) = (json.loads(self.request.data["metadata"]), self.request.data["file"])
+
+        notebook = Notebook.objects.get(id=metadata["notebook_id"])
+        if notebook.owner != self.request.user:
+            raise PermissionDenied
+
         f = File.objects.create(
```

Comment:
Let's use more-than-one-character variable names please. 🌞

Reply:
I think I still have a hangover from python2, where `file` was a reserved keyword...

Reply:
*totally* get it!